jobs/types/user_model_errors_test.py

```diff
@@ -0,0 +1,296 @@
+
+    def test_message_raises_type_error_if_assigned_a_non_string_value(self):
+        class ErrorWithIntMessage(user_model_errors.BaseAuditError):
+            """Subclass that tries to assign an int value to self.message."""
+
+            def __init__(self, model):
+                super(ErrorWithIntMessage, self).__init__(model)
+                self.message = 123
+
+        self.assertRaisesRegexp(
+            TypeError, 'self.message must be a string',
+            lambda: ErrorWithIntMessage(self.model))
+
+    def test_message_raises_value_error_if_assigned_an_empty_value(self):
+        class ErrorWithEmptyMessage(user_model_errors.BaseAuditError):
+            """Subclass that tries to assign an empty value to self.message."""
+
+            def __init__(self, model):
+                super(ErrorWithEmptyMessage, self).__init__(model)
+                self.message = ''
+
+        self.assertRaisesRegexp(
+            ValueError, 'self.message must be a non-empty string',
+            lambda: ErrorWithEmptyMessage(self.model))
+
+    def test_str(self):
+        self.assertEqual(
+            repr(FooError(self.model)),
+            'u"FooError in BaseModel(id=\'123\'): foo"')
+        self.assertEqual(
+            python_utils.UNICODE(FooError(self.model)),
+            'u"FooError in BaseModel(id=\'123\'): foo"')
+
+    def test_equality_between_different_types(self):
+        self.assertNotEqual(FooError(self.model), BarError(self.model))
+
+    def test_equality_between_same_types_and_same_values(self):
+        self.assertEqual(
+            FooError(self.model),
+            FooError(job_utils.clone_model(self.model)))
+
+    def test_equality_between_same_types_and_different_values(self):
+        self.assertNotEqual(
+            FooError(self.model),
+            FooError(job_utils.clone_model(self.model, id='987')))
+
+    def test_hashable(self):
+        set_of_errors = {
+            FooError(self.model),
+            FooError(job_utils.clone_model(self.model)),
+        }
+        self.assertEqual(len(set_of_errors), 1)
+
+    def test_pickling_base_class_raises_not_implemented_error(self):
+        self.assertRaisesRegexp(
+            NotImplementedError,
+            'self.message must be assigned a value in __init__',
+            lambda: pickle.dumps(user_model_errors.BaseAuditError(self.model)))
+
+    def test_pickling_sub_classes(self):
+        foo_error, bar_error = FooError(self.model), BarError(self.model)
+
+        pickled_foo_error, pickled_bar_error = (
+            pickle.dumps(foo_error), pickle.dumps(bar_error))
+        unpickled_foo_error, unpickled_bar_error = (
+            pickle.loads(pickled_foo_error), pickle.loads(pickled_bar_error))
+
+        self.assertEqual(foo_error, unpickled_foo_error)
+        self.assertEqual(bar_error, unpickled_bar_error)
+        self.assertNotEqual(unpickled_foo_error, unpickled_bar_error)
+
+
+class ModelExpiringErrorTests(AuditErrorsTestBase):
```

Comment:
@vojtechjelinek  this test is causing error like this:
![Screenshot from 2021-05-04 14-43-57](https://user-images.githubusercontent.com/46648172/116986028-ec32a980-acea-11eb-82fc-3580d472336b.png)

On backtracking I figured that [this](https://github.com/oppia/oppia/blob/cc565de098c3fba95355fa01db40769b36c4d4aa/jobs/decorators/audit_decorators.py#L123) call in the `decorators.audit_decorator` is causing the issue. Do let me know how I should proceed with the refactoring of the audit_decorators. Also I'm not sure how the decorators are called in the file, so if you have some idea do let me know. Thanks!

Reply:
for now I have added a new file named `user_model_audit_decorator`, but it is just a minor change so I don't want to make a new file for each case

Reply:
@brianrodri Do you have any idea how to fix this?

Reply:
I'm sad this doesn't work out of the box, but since it doesn't let's just remove the typehints as a fix. It isn't a strict requirement anyway, just a convenience.